azurelinuxagent/common/protocol/util.py

```diff
@@ -188,7 +188,7 @@ def _clear_wireserver_endpoint(self):
                 return
             logger.error("Failed to clear wiresever endpoint: {0}", e)
 
-    def _detect_protocol(self, init_goal_state=True):
+    def _detect_protocol(self, init_goal_state=True, save_to_history=False):
```

Comment:
since this method defined as internal use and only not called outside of this. Shouldn't we take parameter from calling method instead default?

Reply:
good point, i removed the default value from this method